common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -523,7 +536,30 @@ public static void beginInteractive(final InteractiveTokenCommand command) {
                 );
             }
 
-            sInteractiveExecutor.execute(new Runnable() {
+            // It is possible that the current task could be stuck irretrievably.  If that's
+            // what's happened, do everything in our power to make certain that it is dead.
+            Future<?> currentInteractiveTask = sCurrentInteractiveTask.get();
+            if (currentInteractiveTask != null && !(currentInteractiveTask.isDone() || currentInteractiveTask.isCancelled())) {
+                try {
+                    // We'll give it 1/2 second to respond to the kill message we broadcast.
+                    currentInteractiveTask.get(500, TimeUnit.MILLISECONDS);
```

Comment:
Who is sending this broadcast? from where?

Reply:
This method does, if the command is a broker command, on line 521.  This raises an interesting question - what do we think we do right now for users who re-entrantly submit interactive tasks? This could perhaps only be done for the broker, but right now it looks like all interactive tasks enqueue with no limit.